domain/operation/service/query.go

```diff
@@ -20,3 +22,19 @@ func (s *Service) GetOperations(ctx context.Context, params operation.QueryArgs)
 func (s *Service) GetOperationByID(ctx context.Context, operationID string) (operation.OperationInfo, error) {
 	return operation.OperationInfo{}, errors.New("actions in Dqlite not supported")
 }
+
+// GetMachineTaskIDsWithStatus retrieves the task IDs for a specific machine name and status.
+func (s *Service) GetMachineTaskIDsWithStatus(ctx context.Context,
+	name coremachine.Name, running corestatus.Status) ([]string, error) {
```

Comment:
thought: you can validate the status as one for operations and tasks with `if running.KnownTaskStatus() {}`